File: packages/hydra-python/src/main/haskell/Hydra/Sources/Python/Manifest.hs
-- | Package manifest for hydra-python.
--
-- Owns the Python coder DSL sources. See feature_290_packaging-plan.md,
-- "Sync system redesign / Package manifests".

module Hydra.Sources.Python.Manifest (
  mainModules,
  testModules,
  dslTypeModules,
) where

import Hydra.Kernel

import qualified Hydra.Sources.Python.Coder as PythonCoder
import qualified Hydra.Sources.Python.Environment as PyEnvironmentSource
import qualified Hydra.Sources.Python.Language as PyLanguage
import qualified Hydra.Sources.Python.Names as PyNames
import qualified Hydra.Sources.Python.Serde as PySerde
import qualified Hydra.Sources.Python.Syntax as PySyntax
import qualified Hydra.Sources.Python.Testing as PythonTesting
import qualified Hydra.Sources.Python.Utils as PyUtils

mainModules :: [Module]
mainModules = [
  PythonCoder.module_,
  PyEnvironmentSource.module_,
  PyLanguage.module_,
  PyNames.module_,
  PySerde.module_,
  PySyntax.module_,
  PythonTesting.module_,
  PyUtils.module_]

-- | Modules in this package whose type definitions should produce derived
-- DSL wrapper modules (Hydra/Dsl/Python/<Name>.hs). The DSL generator
-- consumes this list to emit constructors / accessors / withXxx updaters
-- for each TypeDefinition. Term-only modules (Coder, Serde, Language,
-- Names, Testing, Utils) are deliberately excluded. Extend the list when
-- a new type-defining module needs DSL wrappers.
dslTypeModules :: [Module]
dslTypeModules = [
  PyEnvironmentSource.module_,
  PySyntax.module_]

testModules :: [Module]
testModules = []

File: packages/hydra-python/src/main/python/hydra/sources/python/manifest.py
"""Package manifest for hydra-python.

Mirror of packages/hydra-python/src/main/haskell/Hydra/Sources/Python/Manifest.hs.
Owns the Python coder DSL sources.

This module exports `main_modules` and `test_modules` lists, used by build tooling
to enumerate which Hydra modules belong to the hydra-python package. There is no
per-module JSON output; the manifest contributes to dist/json/hydra-python/manifest.json.
"""

from hydra.packaging import Module

# Imports are deferred to the body of the lists to avoid forcing every module to be built
# when only manifest metadata is needed. Each is the `module_` value from the corresponding
# hydra.sources.python.<X> module.

import hydra.sources.python.coder as _coder
import hydra.sources.python.environment as _environment
import hydra.sources.python.language as _language
import hydra.sources.python.names as _names
import hydra.sources.python.serde as _serde
import hydra.sources.python.syntax as _syntax
import hydra.sources.python.testing as _testing
import hydra.sources.python.utils as _utils


main_modules: list[Module] = [
    _coder.module_,
    _environment.module_,
    _language.module_,
    _names.module_,
    _serde.module_,
    _syntax.module_,
    _testing.module_,
    _utils.module_,
]

test_modules: list[Module] = []
